File: programmers/박소민/lv2/오픈채팅방.js
function solution(record) {
  var answer = [];
  const names = {};

  for (const str of record) {
    const arr = str.split(" ");
    const input = arr[0];
    const id = arr[1];

    if (arr.length === 2) {
      continue;
    }

    const name = arr[2];
    names[id] = name;
  }

  for (const str of record) {
    const arr = str.split(" ");
    const input = arr[0];
    const id = arr[1];

    if (arr.length === 2) {
      answer.push(`${names[id]}님이 나갔습니다.`);
      continue;
    }

    if (input === "Enter") {
      answer.push(`${names[id]}님이 들어왔습니다.`);
    }
  }
  return answer;
}

File: programmers/박소민/lv2/오픈채팅방.py
def solution(record):
    answer = []
    people = {}
    
    for line in record:
        parts = line.split(" ")
        
        # if in 자주 사용하자
        if parts[0] in ("Enter", "Change"):
            people[parts[1]] = parts[2]
            
    for line in record:
        parts = line.split(" ")
        action = parts[0]
        uid = parts[1]
        
        if action == "Enter":
            answer.append(f"{people[uid]}님이 들어왔습니다.")
        elif action == "Leave":
            answer.append(f"{people[uid]}님이 나갔습니다.")
    
    return answer